operator/internal/handlers/internal/storage/secrets.go

```diff
@@ -416,17 +425,22 @@ func extractS3ConfigSecret(s *corev1.Secret, credentialMode lokiv1.CredentialMod
 		if len(roleArn) != 0 {
 			return nil, fmt.Errorf("%w: %s", errSecretFieldNotAllowed, storage.KeyAWSRoleArn)
 		}
+
 		// In the STS case region is not an optional field
 		if len(region) == 0 {
 			return nil, fmt.Errorf("%w: %s", errSecretMissingField, storage.KeyAWSRegion)
 		}
-
 		return cfg, nil
 	case lokiv1.CredentialModeStatic:
 		cfg.Endpoint = string(endpoint)
 
 		if len(endpoint) == 0 {
 			return nil, fmt.Errorf("%w: %s", errSecretMissingField, storage.KeyAWSEndpoint)
+		} else if len(region) != 0 {
+			err = validateS3Endpoint(string(endpoint), string(region))
+			if err != nil {
+				return nil, err
+			}
```

Comment:
Just trying to make it more readable WDYT?
```suggestion
		} 
		if len(region) != 0 {
		   if err := validateS3Endpoint(string(endpoint), string(region)); err != nil {
		      return nil, err
		   }
	        }
```